tests/test_pipeline_media.py

```diff
@@ -129,7 +129,7 @@ def test_should_remove_req_res_references_before_caching_the_results(self):
             try:
                 # Simulate the media_downloaded callback raising a FileException
                 # This usually happens when the status code is not 200 OK
-                raise FileException('download-error') from def_gen_return_exc
+                raise FileException('download-error')
```

Comment:
I think you can use [six.raise_from](https://six.readthedocs.io/index.html?highlight=raise_from#six.raise_from) instead

Reply:
Thanks, but I believe it's not needed. According to [PEP 3134](https://www.python.org/dev/peps/pep-3134/), the `__context__` attribute is automatically set when a chained exception is raised within a try/except block.

I've also made additional changes on Python 2.7 failures that had been suppressed by the huge amount of failing tests.

Reply:
fair point 👍